client/src/tests/modules/auth/container/index.test.jsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { shallow } from "enzyme";
+import { SignIn, mapStateToProps } from "../../../../modules/Auth/container";
+import { props, responses, expected } from "../authMockData";
+
+describe("<Testing the redux container />", () => {
+  it("should mount without crashing", () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+
+  it("should test for returned errors on authentication ", () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().componentWillReceiveProps(props.containerPropsErrors);
+    const actual = wrapper.instance().state;
+    expect(actual).toEqual(expected.notSuccessful);
+  });
+
+
+  it("should test for returned messages on authentication ", () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().componentWillReceiveProps(props.containerPropsMessage);
+    const actual = wrapper.instance().state;
+    expect(actual).toEqual(expected.withMessage);
+  });
+
+
+  it("should test for successful authentication ", () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().componentWillReceiveProps(props.containerPropsData);
+    const actual = props.containerPropsData.history.push;
+    expect(actual).toHaveBeenCalled();
+  });
+
+
+  it("should call the complete signUp method", () => {
+    const spy = jest.fn();
+    const propsNew = { signUp: spy };
+    const mockWrapper = shallow(<SignIn
+      {...props.containerProps}
+      {...propsNew} />);
+    mockWrapper.instance().completeSignUp({ preventDefault: jest.fn() });
+    expect(spy).toHaveBeenCalled();
+  });
+
+
+  it('Should call the onChange method', () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().handleChange({ target: { name: 'name', value: 'victor' } });
+    const actual = wrapper.instance().state.name;
+    expect(actual).toEqual('victor');
+  });
+
+
+  it('Should call the handleSuccess method', () => {
+    const spy = jest.fn();
+    const propsNew = { signIn: spy };
+    const mockWrapper = shallow(<SignIn
+      {...props.containerProps}
+      {...propsNew} />);
+    mockWrapper.instance().handleSuccess(responses.success);
+    expect(spy).toHaveBeenCalled();
+  });
+
+
+  it('Should call the handleSuccess method', () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().handleSuccess(responses.error);
+    expect(wrapper).toMatchSnapshot();
```

Comment:
This function calls state and sets the `data` property of the state object. What do you think about testing that the state is correctly set here as opposed to testing `snapshot`

Reply:
Thanks for the suggestion, the reason behind that was that the state is not being set when testing for errors inside the `handleSuccess` method, therefore no need to check the state.